Remove dead code since sep 2024

Carreau opened the pull request.

Carreau committed:
Remove dead code since sep 2024

peytondmurray reviewed (dismissed):
There are still a few other places where this is being used:

```bash
$ rg 'PYFLYBY_KNOWN_IMPORTS_PATH|PYFLYBY_MANDATORY_IMPORTS_PATH|SUPPORT_DEPRECATED_BEHAVIOR'
conftest.py
64:os.environ["PYFLYBY_KNOWN_IMPORTS_PATH"] = ""
65:os.environ["PYFLYBY_MANDATORY_IMPORTS_PATH"] = ""

lib/python/pyflyby/_py.py
1837:                os.environ["PYFLYBY_KNOWN_IMPORTS_PATH"] = ""
1838:                os.environ["PYFLYBY_MANDATORY_IMPORTS_PATH"] = ""

lib/python/pyflyby/_importdb.py
317:                os.getenv("PYFLYBY_KNOWN_IMPORTS_PATH"),
318:                os.getenv("PYFLYBY_MANDATORY_IMPORTS_PATH"),
334:                               os.getenv("PYFLYBY_KNOWN_IMPORTS_PATH"),
335:                               os.getenv("PYFLYBY_MANDATORY_IMPORTS_PATH")))
```

Carreau committed:
Cascade removal of deprecated items

This triggers the removal of a number of unused and deprecated
parameters from function, up to the replacement of the single
_from_args call to a _from_code, and supression of _from_args, which is
not used anymore.
